### PR TITLE
InteractionManagers

### PR DESCRIPTION
**Pull request by danilwhale**
Finishes mapping of `*InteractionManager` and adds `TestInteractionManager`

### CODE REVIEW

File: mappings/net/minecraft/client/InteractionManager.mapping [hunk 1]
```diff
@@ -1,17 +1,60 @@
 CLASS net/minecraft/class_504 net/minecraft/client/InteractionManager
 	FIELD field_2104 minecraft Lnet/minecraft/client/Minecraft;
+	FIELD field_2105 noTick Z
 	METHOD <init> (Lnet/minecraft/client/Minecraft;)V
 		ARG 1 minecraft
+	METHOD method_1705 resetBlockBreaking ()V
+	METHOD method_1706 update (F)V
+	METHOD method_1707 interactBlock (IIII)V
```
Comment (DanyGames2014) [inline mappings/net/minecraft/client/InteractionManager.mapping]: [attackBlock](https://maven.fabricmc.net/docs/yarn-1.21.1+build.3/net/minecraft/client/network/ClientPlayerInteractionManager.html#attackBlock(net.minecraft.util.math.BlockPos,net.minecraft.util.math.Direction)) in Yarn

File: mappings/net/minecraft/client/InteractionManager.mapping [hunk 2]
```diff
@@ -1,17 +1,60 @@
 CLASS net/minecraft/class_504 net/minecraft/client/InteractionManager
 	FIELD field_2104 minecraft Lnet/minecraft/client/Minecraft;
+	FIELD field_2105 noTick Z
 	METHOD <init> (Lnet/minecraft/client/Minecraft;)V
 		ARG 1 minecraft
+	METHOD method_1705 resetBlockBreaking ()V
```
Comment (DanyGames2014) [inline mappings/net/minecraft/client/InteractionManager.mapping]: [cancelBlockBreaking](https://maven.fabricmc.net/docs/yarn-1.21.1+build.3/net/minecraft/client/network/ClientPlayerInteractionManager.html#cancelBlockBreaking()) in Yarn

File: mappings/net/minecraft/client/InteractionManager.mapping [hunk 3]
```diff
@@ -1,17 +1,60 @@
...
+		ARG 2 y
+		ARG 3 z
+		ARG 4 direction
 	METHOD method_1708 clickSlot (IIIZLnet/minecraft/class_54;)Lnet/minecraft/class_31;
 		ARG 1 windowId
 		ARG 2 slotId
 		ARG 3 button
 		ARG 4 shift
 		ARG 5 player
+	METHOD method_1709 onContainerRemove (ILnet/minecraft/class_54;)V
```
Comment (DanyGames2014) [inline mappings/net/minecraft/client/InteractionManager.mapping]: Container is ScreenHandler in BINY. Altho this is called purely from HandledScreen, not ScreenHandler so something like `onScreenRemoved(syncId, player)`  would be more accurate

File: mappings/net/minecraft/client/InteractionManager.mapping [hunk 4]
```diff
@@ -1,17 +1,60 @@
...
 		ARG 1 minecraft
+	METHOD method_1705 resetBlockBreaking ()V
+	METHOD method_1706 update (F)V
+	METHOD method_1707 interactBlock (IIII)V
+		ARG 1 x
+		ARG 2 y
+		ARG 3 z
+		ARG 4 direction
 	METHOD method_1708 clickSlot (IIIZLnet/minecraft/class_54;)Lnet/minecraft/class_31;
 		ARG 1 windowId
```
Comment (DanyGames2014) [inline mappings/net/minecraft/client/InteractionManager.mapping]: While this isnt part of the PR, it feels fitting to correct this to syncId while we are here

File: mappings/net/minecraft/client/InteractionManager.mapping [hunk 5]
```diff
@@ -1,17 +1,60 @@
...
+		ARG 2 player
+	METHOD method_1710 setWorld (Lnet/minecraft/class_18;)V
+		ARG 1 world
+	METHOD method_1711 preparePlayer (Lnet/minecraft/class_54;)V
+		ARG 1 player
+	METHOD method_1712 interactItem (Lnet/minecraft/class_54;Lnet/minecraft/class_18;Lnet/minecraft/class_31;)Z
+		ARG 1 player
+		ARG 2 world
+		ARG 3 item
+	METHOD method_1713 interactBlockWithItem (Lnet/minecraft/class_54;Lnet/minecraft/class_18;Lnet/minecraft/class_31;IIII)Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/client/InteractionManager.mapping]: [interactBlock](https://maven.fabricmc.net/docs/yarn-1.20-pre4+build.2/net/minecraft/server/network/ServerPlayerInteractionManager.html#interactBlock(net.minecraft.server.network.ServerPlayerEntity,net.minecraft.world.World,net.minecraft.item.ItemStack,net.minecraft.util.Hand,net.minecraft.util.hit.BlockHitResult)) in Yarn

File: mappings/net/minecraft/client/InteractionManager.mapping [hunk 6]
```diff
@@ -1,17 +1,60 @@
...
+		ARG 4 direction
+	METHOD method_1717 createPlayer (Lnet/minecraft/class_18;)Lnet/minecraft/class_54;
+		ARG 1 world
+	METHOD method_1718 preparePlayerRespawn (Lnet/minecraft/class_54;)V
+		ARG 1 player
 	METHOD method_1719 attackEntity (Lnet/minecraft/class_54;Lnet/minecraft/class_57;)V
 		ARG 1 player
 		ARG 2 target
 	METHOD method_1720 tick ()V
+	METHOD method_1721 onBlockBreakingAction (IIII)V
```
Comment (DanyGames2014) [inline mappings/net/minecraft/client/InteractionManager.mapping]: [processBlockBreakingAction](https://maven.fabricmc.net/docs/yarn-1.20-pre4+build.2/net/minecraft/server/network/ServerPlayerInteractionManager.html#processBlockBreakingAction(net.minecraft.util.math.BlockPos,net.minecraft.network.packet.c2s.play.PlayerActionC2SPacket.Action,net.minecraft.util.math.Direction,int,int)) in Yarn

File: mappings/net/minecraft/MultiplayerInteractionManager.mapping [hunk 7]
```diff
@@ -1,2 +1,15 @@
 CLASS net/minecraft/class_608 net/minecraft/MultiplayerInteractionManager
+	FIELD field_2608 miningPosX I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/MultiplayerInteractionManager.mapping]: maybe `breakingPosX` so we have a consistent terminology and dont switch between mining and breaking (like yarn (: ), technically `currentBreakingPosX` would be the closest in that case, but that feels obscenely long. Block breaking feels more appropriate then mining imo

File: mappings/net/minecraft/MultiplayerInteractionManager.mapping [hunk 8]
```diff
@@ -1,2 +1,15 @@
...
+	FIELD field_2609 miningPosY I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/MultiplayerInteractionManager.mapping]: Same comment as miningPosX

File: mappings/net/minecraft/MultiplayerInteractionManager.mapping [hunk 9]
```diff
@@ -1,2 +1,15 @@
...
+	FIELD field_2610 miningPosZ I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/MultiplayerInteractionManager.mapping]: Same comment as miningPosX

File: mappings/net/minecraft/MultiplayerInteractionManager.mapping [hunk 10]
```diff
@@ -1,2 +1,15 @@
...
+	FIELD field_2611 blockBreakingProgress F
+	FIELD field_2612 lastBlockBreakingProgress F
+	FIELD field_2613 miningSoundDelayTicks F
```
Comment (DanyGames2014) [inline mappings/net/minecraft/MultiplayerInteractionManager.mapping]: Same comment as miningPosX

File: mappings/net/minecraft/MultiplayerInteractionManager.mapping [hunk 11]
```diff
@@ -1,2 +1,15 @@
...
+	FIELD field_2614 miningDelayTicks I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/MultiplayerInteractionManager.mapping]: Same comment as miningPosX

File: mappings/net/minecraft/MultiplayerInteractionManager.mapping [hunk 12]
```diff
@@ -1,2 +1,15 @@
...
+	FIELD field_2615 mining Z
```
Comment (DanyGames2014) [inline mappings/net/minecraft/MultiplayerInteractionManager.mapping]: Same comment as miningPosX.  `breakingBlock` would be the closest

File: mappings/net/minecraft/SingleplayerInteractionManager.mapping [hunk 13]
```diff
@@ -1 +1,8 @@
 CLASS net/minecraft/class_520 net/minecraft/SingleplayerInteractionManager
+	FIELD field_2181 miningPosX I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/SingleplayerInteractionManager.mapping]: Same as MultiplayerInteractionManager

File: mappings/net/minecraft/SingleplayerInteractionManager.mapping [hunk 14]
```diff
@@ -1 +1,8 @@
...
+	FIELD field_2182 miningPosY I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/SingleplayerInteractionManager.mapping]: Same as MultiplayerInteractionManager

File: mappings/net/minecraft/SingleplayerInteractionManager.mapping [hunk 15]
```diff
@@ -1 +1,8 @@
...
+	FIELD field_2183 miningPosZ I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/SingleplayerInteractionManager.mapping]: Same as MultiplayerInteractionManager

File: mappings/net/minecraft/SingleplayerInteractionManager.mapping [hunk 16]
```diff
@@ -1 +1,8 @@
...
+	FIELD field_2184 blockBreakingProgress F
+	FIELD field_2185 lastBlockBreakingProgress F
+	FIELD field_2186 miningSoundDelayTicks F
```
Comment (DanyGames2014) [inline mappings/net/minecraft/SingleplayerInteractionManager.mapping]: Same as MultiplayerInteractionManager

File: mappings/net/minecraft/SingleplayerInteractionManager.mapping [hunk 17]
```diff
@@ -1 +1,8 @@
...
+	FIELD field_2187 miningDelayTicks I
```
Comment (DanyGames2014) [inline mappings/net/minecraft/SingleplayerInteractionManager.mapping]: Same as MultiplayerInteractionManager

**Review by DanyGames2014 — CHANGES_REQUESTED** [review]

**Review by DanyGames2014 — APPROVED** [review]
Vincent, we happy?
Yeah, we're happy

**Comment by matthewperiut** [comment]
It seems that the javadoc lint task fails due to something in this commit, see log:
```
> Task :javadocLint FAILED
lint: net/minecraft/screen/ScreenHandlerListener.onPropertyUpdate(Lnet/minecraft/class_71;II)V.syncId: parameter javadoc ends with '.'
lint: net/minecraft/screen/ScreenHandlerListener.onPropertyUpdate(Lnet/minecraft/class_71;II)V.syncId: parameter javadoc starts with uppercase word 'Arbitrary'
lint: net/minecraft/screen/ScreenHandlerListener.onPropertyUpdate(Lnet/minecraft/class_71;II)V.trackedValue: parameter javadoc ends with '.'
lint: net/minecraft/screen/ScreenHandlerListener.onPropertyUpdate(Lnet/minecraft/class_71;II)V.trackedValue: parameter javadoc starts with uppercase word 'This'
```

Apparently, these are considered (later in the log it says) javadoc format errors:
```
 What went wrong:
Execution failed for task ':javadocLint'.
 A failure occurred while executing net.fabricmc.filament.task.JavadocLintTask$LintAction
    Found 4 javadoc format errors! See the log for details.
```

**Review by calmilamsy — CHANGES_REQUESTED** [review]
Please fix the javadoc issues.

**Comment by matthewperiut** [comment]
gg javadoc issue was mine, sorry about that guys

**Review by calmilamsy — APPROVED** [review]
Javadoc was unrelated to this PR, merging.

**Pull request merged**